promql/functions.go

```diff
@@ -1271,87 +1247,56 @@ func funcPredictLinear(vals []parser.Value, args parser.Expressions, enh *EvalNo
 	return append(enh.Out, Sample{F: slope*duration + intercept}), nil
 }
 
-// === histogram_count(Vector parser.ValueTypeVector) (Vector, Annotations) ===
-func funcHistogramCount(vals []parser.Value, _ parser.Expressions, enh *EvalNodeHelper) (Vector, annotations.Annotations) {
-	inVec := vals[0].(Vector)
-
-	for _, sample := range inVec {
-		// Skip non-histogram samples.
-		if sample.H == nil {
-			continue
-		}
-		if !enh.enableDelayedNameRemoval {
-			sample.Metric = sample.Metric.DropMetricName()
+func simpleHistFunc(vals []parser.Value, enh *EvalNodeHelper, f func(h *histogram.FloatHistogram) float64) Vector {
```

Comment:
I'm not a great fan of abbreviations that do not abbreviate much but still create ambiguity (has this to do with historic values?).

Let's call this `simpleHistogramFunc` or if we want to be short, do `simpleHFunc`.

Maybe we should also rename the existing `simpleFunc` to `simpleFloatFunc` or `simpleFFunc`.

My personal preference would be `simpleHistogramFunc` and `simpleFloatFunc`.

Reply:
Yeah, `simpleHistogramFunc` and `simpleFloatFunc` sound better. I'll change it as soon as possible.